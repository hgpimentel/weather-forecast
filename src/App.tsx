import { Route, Switch, Redirect } from "react-router-dom";
import ROUTE from "./config/routes";
import { Today, Daily, NotFound } from "./pages";
import { Layout } from "./components";

const App = () => {
  return (
    <Layout>
      <Switch>
        <Route exact path="/">
          <Redirect to={ROUTE.TODAY} />
        </Route>
        <Route path={ROUTE.TODAY}>
          <Today />
        </Route>
        <Route path={ROUTE.NEXT7DAYS}>
          <Daily />
        </Route>
        <Route path="*">
          <NotFound message="Page not found!" />
        </Route>
      </Switch>
    </Layout>
  );
};

export default App;
