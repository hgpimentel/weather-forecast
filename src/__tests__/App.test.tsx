import App from "../App";
import renderer from "react-test-renderer";
import { BrowserRouter } from "react-router-dom";
import { render, screen, fireEvent } from "@testing-library/react";

describe("Select location component", () => {
  it("renders correctly", () => {
    const component = renderer
      .create(
        <BrowserRouter>
          <App />
        </BrowserRouter>
      )
      .toJSON();

    expect(component).toMatchSnapshot();
  });

  it("has header title", () => {
    render(
      <BrowserRouter>
        <App />
      </BrowserRouter>
    );

    expect(screen.getByRole("heading")).toHaveTextContent(
      "Weather Forecast App"
    );
  });

  it("has today page active", () => {
    render(
      <BrowserRouter>
        <App />
      </BrowserRouter>
    );

    expect(screen.getByText("Today")).toHaveClass("active");
  });
});
