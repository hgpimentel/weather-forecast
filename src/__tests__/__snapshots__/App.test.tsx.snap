// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select location component renders correctly 1`] = `
Array [
  <header
    className="sc-fujyAs fGuyzn"
  >
    <div
      className="sc-bdnxRM sc-jrsJWt eoiijl iedUoR"
    >
      <h1>
        Weather Forecast App
      </h1>
    </div>
    <nav
      className="sc-pNWdM ljLxKa"
    >
      <ul>
        <li>
          <a
            aria-current="page"
            className="active"
            href="/today"
            onClick={[Function]}
            style={Object {}}
          >
            Today
          </a>
        </li>
        <li>
          <a
            aria-current={null}
            href="/next7days"
            onClick={[Function]}
          >
            Next 7 Days
          </a>
        </li>
      </ul>
    </nav>
  </header>,
  <main />,
]
`;
